tests/cpp/parser_tests.cpp

```diff
@@ -2299,6 +2341,110 @@ rule same_variable_in_nested_for_loops {
 	}
 }
 
+TEST_F(ParserTests,
+AndroguardModuleWorks) {
+	prepareInput(
+R"(
+import "androguard"
+
+rule dummy_rule {
+	condition:
+		androguard.max_sdk > androguard.signature.hits("dummy") and androguard.min_sdk == androguard.max_sdk
+}
+)");
+
+	EXPECT_TRUE(driver.parse());
+	ASSERT_EQ(1u, driver.getParsedFile().getRules().size());
+
+	const auto& rule = driver.getParsedFile().getRules()[0];
+	EXPECT_EQ(R"(androguard.max_sdk > androguard.signature.hits("dummy") and androguard.min_sdk == androguard.max_sdk)", rule->getCondition()->getText());
+
+	EXPECT_EQ(input_text, driver.getParsedFile().getTextFormatted());
+}
+
+TEST_F(ParserTests,
+AndroguardModuleUnrecognized) {
+	prepareInput(
+R"(
+import "androguard"
+
+rule dummy_rule {
+	condition:
+		androguard.max_sdk > androguard.signature.hits("dummy")
+}
+)");
+
+	Module::reset("androguard");
```

Comment:
Will user need to call `Module::reset()` manually if they want to create more instances of yaramod? That doesn't sound very user friendly. This needs to happen all under the hood without user being aware of that. Please, change it so that user won't need to do this.

Reply:
The user does not need to call the `reset` method, we only need it in our tests where we construct more ParserDriver instances - all sharing the same modules, because modules are stored as global variables in `module.cpp`. These modules need to be reseted if we need ParserDriver with different `Features`. This is something that the user will never need, because they will use the same `Features` among their `ParserDrivers`. Maybe we could stop storing modules as global variables but store them either inside `Yaramod` or inside `ParserDriver` classes.

Reply:
But what if I want to create `Yaramod` instance once with VirusTotal features and once with no additional features enabled? Would I need to do that?

Reply:
No, that would not require you calling the `reset` method. However, if you decided to use two different `Yaramod` instances, first one with `ImportFeature::Avast` and then second one with `ImportFeature::Basic` then yes, you would have to call the static method `Module::reset(<module_name>)`. The problem is, that both `ParserDriver` instances share the same modules. In my opinien, it is a good thing that the modules are shared between the instances. The whole mean of issue #46 was to reduce the number of symbols stored and the complexity of modules. Therefore I propose a solution which would keep it:

Along with the globally stored modules we will store also global `Feature` instance named `modules_state`. Whenever new instance of `Yaramod` with parameter `ImportFeature feature` is created, we look at `modules_state` and update it:
`modules_state |= feature`
possibly enabling new modules/symbols. With this approach, when loading a module, we will only look at the global variable `modules_state`.

What do you think?

Reply:
> However, if you decided to use two different Yaramod instances, first one with ImportFeature::Avast and then second one with ImportFeature::Basic then yes, you would have to call the static method Module::reset(<module_name>). 

This is what I meant and I don't personally like it because it is so easy to misuse from user standpoint.

>  In my opinien, it is a good thing that the modules are shared between the instances.

Why do you think it is a good idea to keep the modules shared? I personally think that it was good idea when modules were immutable but I don't see any benefit when modules are possibly mutable.

Reply:
I would propose to move from global modules to modules that are only local to certain `ParserDriver` instance. All the modules will have to be initialized multiple times when creating multiple `Yaramod` instances but I think that benefits of this solution outweigh benefits of having global modules.

Reply:
I thought about it and I agree with you. We will get rid of global variables, we will get rid of the `reset` function or some hacks I proposed. It will be much cleaner and use just a liitle bit more memory to store the modules for each instance independently. I will implement it soon. Thank you for this discussion.

Reply:
Done: In [commit](https://github.com/avast/yaramod/pull/51/commits/ef2a4f3fb3eb01b73404fcc84284b5ec751cc60d) you will find a `ModulesPool` class which stores the modules. Each `ParserDriver` or `YaraFileBuilder` is assigned one instance of the `ModulesPool` class, which holds all used modules.

Reply:
Done: In [commit](https://github.com/avast/yaramod/pull/51/commits/7afaecccd237d1cbc8e159bd8217962d7d3009c3) we use new class `ModulesPool` which stores the modules. Each `ParserDriver` or `YaraFileBuilder` instance is assigned its own instance of `ModulesPool` class.